4.gene_expression_signatures/scripts/2.download-target-data.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Download TARGET Pan Cancer Data for Compression
+# The notebook downloads gene expression and clinical data from the TARGET project. The data is downloaded from UCSC Xena.
+# The data is in log2(FPKM) RSEM transformed
+
+# In[1]:
+
+
+import os
+from urllib.request import urlretrieve
+import random
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+# In[2]:
+
+
+url = 'https://toil-xena-hub.s3.us-east-1.amazonaws.com/download/'
+name = 'target_RSEM_gene_fpkm.gz'
+
+path = os.path.join('data', name)
+
+
+# In[3]:
+
+
+urlretrieve('{}{}'.format(url, name), path)
+
+
+# In[4]:
+
+
+get_ipython().system(" sha256sum 'data/target_RSEM_gene_fpkm.gz'")
+
+
+# In[5]:
+
+
+name = 'gencode.v23.annotation.gene.probemap'
+
+path = os.path.join('data', name)
+
+
+# In[6]:
+
+
+urlretrieve('{}probeMap/{}'.format(url, name), path)
+
+
+# In[7]:
+
+
+get_ipython().system(" sha256sum 'data/gencode.v23.annotation.gene.probemap'")
+
+
+# In[8]:
+
+
+name = 'TARGET_phenotype.gz'
+
+path = os.path.join('data', name)
+
+
+# In[9]:
+
+
+urlretrieve('{}{}'.format(url, name), path)
+
+
+# In[10]:
+
+
+get_ipython().system(" sha256sum 'data/TARGET_phenotype.gz'")
+
+
+# # Process TARGET PanCancer Data
+# Retrive the downloaded expression data, update gene identifiers to entrez, and curate sample IDs. The script will also identify a balanced hold-out test set to compare projection performance into learned latent spaces across algorithms. 
+
+# In[11]:
+
+
+random.seed(1234)
```

Comment:
Consider changing to random seed of 0 for way lab standards

Reply:
Done!